logr.go

```diff
@@ -430,15 +430,18 @@ func (l Logger) valuesFromContext() []interface{} {
 	// pairs, of which each needs two entries.
 	keysAndValues := make([]interface{}, 0, 4*len(*l.contextHandlers))
 	for _, handler := range *l.contextHandlers {
-		for _, keyAndValue := range handler(l.ctx) {
+		for _, keyAndValue := range handler.ValuesFromContext(l.ctx) {
 			keysAndValues = append(keysAndValues, keyAndValue.Key, keyAndValue.Value)
 		}
 	}
 	return keysAndValues
 }
 
-// ContextHandler retrieves certain values from the context.
-type ContextHandler func(ctx context.Context) KeysAndValues
+// ContextHandler provides callbacks for working with a context.
+// More functionality might get added in the future.
+type ContextHandler struct {
```

Comment:
Maybe this would be how a user would designate cacheable and non-cacheable?

Reply:
Could be, although I would start with just "cacheable" because the use cases for non-cacheable aren't clear.